rust/src/delta.rs

```diff
@@ -627,8 +629,45 @@ impl DeltaTable {
             .get_tombstones()
             .iter()
             .filter(|tombstone| tombstone.deletionTimestamp < delete_before_timestamp)
+            .collect())
```

Comment:
looking at the code in https://github.com/delta-io/delta/blob/34002d3349dcb2cac38fcc44b89d584fffa63751/core/src/main/scala/org/apache/spark/sql/delta/commands/VacuumCommand.scala#L95, looks the right behavior here should be listing all data files pass retention date in a table, then subtract that with files being tracked in delta tables (including removed files) to get to the final list of files to delete.

Reply:
You are right! 

Vacuum steps are the following:
1. List all files inside the DeltaTable root path
2. Filter by subtracting the files still in use and tombstones that don't reach the retention threshold
3. Delete the rest